### ISSUE TITLE
Saule cannot filter more than one input parameter.

### ISSUE THREAD

**Issue by dhirajdhakal09** [joukevandermaas/saule#112]
Hi,

I tried to filter out with two input parameters with comma separated as defined by jsonapi.org but it always returns null.

http://example.com/articles/filter[something]=1234 works

http://example.com/articles/filter[something]=1245,1235 won't work.

is this feature missing?

Thanks

**Comment by joukevandermaas** [comment]
I think it's probably not supported yet, but you might be able to get it working using custom filter expressions. See  [here](https://github.com/joukevandermaas/saule/wiki/Queryable-endpoints#customizing-filtering-expressions) for more info.

I assume the value you'd get is the string including the comma.

**Comment by dhirajdhakal09** [comment]
Hi Jouke,

Has this already in place?

Please let me know.

Thanks!!



On Thu, Jun 2, 2016 at 1:20 PM, Jouke van der Maas <notifications@github.com

**Comment by joukevandermaas** [comment]
It isn't. I'm open to a PR but I don't have time to pick this up ATM.

**Comment by PhyberApex** [comment]
If I wanted to pick this up what would  be a good solution to escape a ","?

~Cheers

**Comment by joukevandermaas** [comment]
@PhyberApex I don't think we need to support escaping commas to be honest. If the spec doesn't mention it, it shouldn't be blocking us IMO.

**Comment by joukevandermaas** [comment]
I suppose we could do like in CSV? So if the filter string is in quotes, we don't interpret the commas?

`?filter[prop]="some,value"` => checks that `prop` is `"some,value"`.
`?filter[prop]=some,value` => checks that `prop` is `some` or `value`.

**Comment by PhyberApex** [comment]
Just so I understand you correctly. This would mean I could not do a "or" filter on multiple strings containing "," is this correct?

~Cheers

**Comment by joukevandermaas** [comment]
It would be like `?filter[prop]="a,b","b,c"`, I think.

**Comment by PhyberApex** [comment]
Okay I will probably take a look at this during this weekend. And open an PR when I'm ready.

~Cheers

**Comment by joukevandermaas** [comment]
@PhyberApex a PR with the first steps would be great, thanks! If I remember right, the filtering code is pretty complex.
